atc/api/resources_test.go

```diff
@@ -1608,4 +1608,260 @@ var _ = Describe("Resources API", func() {
 			})
 		})
 	})
+
+	Describe("DELETE /api/v1/teams/:team_name/pipelines/:pipeline_name/resources/:resource_name/cache", func() {
+		var response *http.Response
+		var fakeResource *dbfakes.FakeResource
+
+		Context("when authenticated ", func() {
+			BeforeEach(func() {
+				fakeAccess.IsAuthenticatedReturns(true)
+			})
+
+			Context("when authorized", func() {
+				BeforeEach(func() {
+					fakeAccess.IsAuthorizedReturns(true)
+				})
+
+				Context("when it tries to find a resource", func() {
+					JustBeforeEach(func() {
+						var err error
+
+						request, err := http.NewRequest("DELETE", server.URL+"/api/v1/teams/a-team/pipelines/a-pipeline/resources/resource-name/cache", nil)
+						Expect(err).NotTo(HaveOccurred())
+
+						response, err = client.Do(request)
+						Expect(err).NotTo(HaveOccurred())
+					})
+
+					It("calls the resource", func() {
+						resourceName := fakePipeline.ResourceArgsForCall(0)
+						Expect(resourceName).To(Equal("resource-name"))
+					})
+				})
+
+				Context("when finding the resource succeeds", func() {
+					BeforeEach(func() {
+						fakeResource = new(dbfakes.FakeResource)
+						fakeResource.IDReturns(1)
+						fakePipeline.ResourceReturns(fakeResource, true, nil)
+					})
+
+					Context("when clear cache succeeds", func() {
```

Comment:
You can actually move the `fakeResource.ClearResourceCacheReturns(1,nil)` on line 1662 to here underneath this context for `"when clear cache succeeds"` because that line for faking out the return value is actually mocking out the situation for when the clear cache succeeds (which is what the context is describing). So it would look something like

```
Context("when clear cache succeeds", func() {
  BeforeEach(func() {
    fakeResource.ClearResourceCacheReturns(1,nil)
  })
})
```

And then the `fakeResource.ClearResourceCacheReturns(1,nil)` line can be removed from line 1662!